### ISSUE TITLE
Sites are blocked or unblocked based on when the "Save Options" button was last pressed, not on the current time.

### ISSUE THREAD

**Issue by rl2939** [ray-lothian/Block-Site#116]
# Problem

Lets define two blocks of time:
Block A: When sites are supposed to be blocked. (ex: 7 pm - 1:59 pm)
Block B: When sites are supposed to be unblocked. (ex: 2 pm - 6:59 pm)

Lets then say that I create a site blocklist and save my changes when Block A is active (ex: 1:05 pm).
When the time changes to where Block B is active (ex: 2:05 pm) and I visit a site in the blocklist, it displays a block notice **as if I visited the site during Block A**.

If I then hit the save button on the options menu during Block B, the sites will become unblocked. However, if I then visit a site on my blocklist when Block A is active (ex: 8 pm), the sites **will still be unblocked** until I go the options menu again and hit the save button (during block A).

## Expected Behavior
The sites are blocked and unblocked _at the times selected in the options menu_.

## Actual Behavior
The sites are blocked or unblocked _based on the time I last hit the block button._

## Other information

Version: 0.4.9
Browser: Firefox

A potential solution may be to run the function that pressing the "Save Options" does every minute in the background.

Related issues: https://github.com/ray-lothian/Block-Site/issues/36 and https://github.com/ray-lothian/Block-Site/issues/39. **It is not a duplicate of these issues, those issues either have the blocklist stuck on blocking or unblocking mode _all the time_, not blocking or unblocking _based on the time the save button is pressed_.**

**Comment by shubhams** [comment]
I can reproduce this problem on macOS!
Version: 0.5.2.1
Browser: Firefox
